yarn-project/merkle-tree/src/snapshots/snapshot_builder_test_suite.ts

```diff
@@ -0,0 +1,138 @@
+import { SnapshotBuilder } from './snapshot_builder.js';
+
+/** Creates a test suit for snapshots */
+export function describeSnapshotBuilderTestSuite<T extends TreeBase, S extends SnapshotBuilder>(
+  getTree: () => T,
+  getSnapshotBuilder: () => S,
+  modifyTree: (tree: T) => Promise<void>,
+) {
+  describe('SnapshotBuilder', () => {
+    let tree: T;
+    let snapshotBuilder: S;
+    let leaves: bigint[];
+
+    beforeEach(() => {
+      tree = getTree();
+      snapshotBuilder = getSnapshotBuilder();
+
+      leaves = Array.from({ length: 4 }).map(() => BigInt(Math.floor(Math.random() * 2 ** tree.getDepth())));
+    });
+
+    describe('snapshot', () => {
+      it('takes snapshots', async () => {
+        await modifyTree(tree);
+        await tree.commit();
+        await expect(snapshotBuilder.snapshot(1)).resolves.toBeDefined();
+      });
+
+      it('refuses to take the same snapshot twice', async () => {
+        await modifyTree(tree);
+        await tree.commit();
+
+        const version = 1;
+        await snapshotBuilder.snapshot(version);
+        await expect(snapshotBuilder.snapshot(version)).rejects.toThrow();
+      });
+
+      it('returns the same path if tree has not advanced', async () => {
+        await modifyTree(tree);
+        await tree.commit();
+        const snapshot = await snapshotBuilder.snapshot(1);
+
+        const historicPaths = await Promise.all(leaves.map(leaf => snapshot.getSiblingPath(leaf, false)));
+        const expectedPaths = await Promise.all(leaves.map(leaf => tree.getSiblingPath(leaf, false)));
+
+        for (const [index, path] of historicPaths.entries()) {
+          expect(path).toEqual(expectedPaths[index]);
+        }
+      });
+
+      it('returns historic paths if tree has diverged and no new snapshots have been taken', async () => {
+        await modifyTree(tree);
+        await tree.commit();
+        const snapshot = await snapshotBuilder.snapshot(1);
+
+        const expectedPaths = await Promise.all(leaves.map(leaf => tree.getSiblingPath(leaf, false)));
+
+        await modifyTree(tree);
+        await tree.commit();
+
+        const historicPaths = await Promise.all(leaves.map(leaf => snapshot.getSiblingPath(leaf, false)));
+
+        for (const [index, path] of historicPaths.entries()) {
+          expect(path).toEqual(expectedPaths[index]);
+        }
+      });
+
+      it('returns historic paths at old snapshots', async () => {
+        await modifyTree(tree);
+        await tree.commit();
+        const snapshot = await snapshotBuilder.snapshot(1);
+
+        const historicPaths = await Promise.all(leaves.map(leaf => snapshot.getSiblingPath(leaf, false)));
+        const expectedPaths = await Promise.all(leaves.map(leaf => tree.getSiblingPath(leaf, false)));
+
+        await modifyTree(tree);
+        await tree.commit();
+
+        await snapshotBuilder.snapshot(2);
+
+        for (const [index, path] of historicPaths.entries()) {
```

Comment:
The values you are matching are both defined before you modify anything so it don't make a lot of difference. Seems like the `historicPaths` should only be computed after the second snapshot to show it. Otherwise the test is the same as `returns the same path if tree has not advanced`.

I think this test can be removed as it is covered be the other ones.

Reply:
Fair, I think I wanted to prove to myself that snapshotting again does not modify any prior sibling paths I had generated. But now that I say it out loud the test indeed doesn't make sense :sweat_smile: